packages/@aws-cdk/aws-lambda/test/function.test.ts

```diff
@@ -2487,6 +2487,16 @@ describe('function', () => {
   });
 });
 
+test('throws if ephemeral storage size is out of bound', () => {
```

Comment:
Can you add 2 more unit tests? 
1. Showing that the property works, just the normal happy path.
2. Showing that you can use an unresolved token as the input for ephemeralStorageSize?

Reply:
@madeline-k  #1 is done... but can you elaborate more on why #2 is needed (learning moment for me). Also, if you could point me to an example test that would be awesome 😃 . TY.

Reply:
Here is an example: https://github.com/aws/aws-cdk/blob/a81a0a45d72bf297c60ad01c7e5f84ea11b8cb61/packages/%40aws-cdk/aws-ssm/test/parameter.test.ts#L87

Reply:
You can do this to create an unresolved token: 
```
cdk.Size.gibibytes(cdk.Lazy.number({ produce: () => 1 }));
```

Reply:
I am not sure why exactly @kaizen3031593 requested it in this case. But in general, we test for this to make sure that validation doesn't prevent users from passing in unresolved tokens. Like CfnOutputs from other stacks.

Reply:
TY!

Reply:
@madeline-k I tried using example you posted but I keep getting this error:
```ts
test('ephemeral storage allows unresolved tokens', () => {
  const stack = new cdk.Stack();
  expect(() => {
    new lambda.Function(stack, 'MyLambda', {
      code: new lambda.InlineCode('foo'),
      handler: 'bar',
      runtime: lambda.Runtime.NODEJS_14_X,
      ephemeralStorageSize:  Size.gibibytes(Lazy.number({ produce: () => 1 })),
    });
  }).not.toThrow();
});
```
```bash
FAIL test/function.test.js (13.295 s)
@aws-cdk/aws-lambda:   ● ephemeral storage allows unresolved tokens
@aws-cdk/aws-lambda:     Unable to perform time unit conversion on un-resolved token -1.8881545897092687e+289.
@aws-cdk/aws-lambda:       191 |   if (fromUnit.inKibiBytes === toUnit.inKibiBytes) { return amount; }
@aws-cdk/aws-lambda:       192 |   if (Token.isUnresolved(amount)) {
@aws-cdk/aws-lambda:     > 193 |     throw new Error(`Unable to perform time unit conversion on un-resolved token ${amount}.`);
@aws-cdk/aws-lambda:           |           ^
@aws-cdk/aws-lambda:       194 |   }
@aws-cdk/aws-lambda:       195 |
@aws-cdk/aws-lambda:       196 |   const multiplier = fromUnit.inKibiBytes / toUnit.inKibiBytes;
@aws-cdk/aws-lambda:       at convert (../core/lib/size.ts:193:11)
@aws-cdk/aws-lambda:       at Size.toMebibytes (../core/lib/size.ts:112:12)
@aws-cdk/aws-lambda:       at new Function (lib/function.ts:754:38)
@aws-cdk/aws-lambda:       at Object.<anonymous> (test/function.test.ts:2526:3)
```
I also saw this test implementation, but that didn't work either: 
https://github.com/aws/aws-cdk/blob/f19896be08209f140fdf07189b9d1e4c9d601bc6/packages/%40aws-cdk/core/test/size.test.ts#L12

I guess I just am not really sure what I'm testing here tbh. :)

Reply:
With an unresolved token, you need to test with something in KibiBytes, since that is the target. @rix0rrr actually just made a change to make this error message much more helpful :)

Reply:
Actually, I think the target should be MebiBytes in this case. So I'm a bit confused why the error you get is from a conversion to KibiBytes.

Reply:
Yup... waiting for https://github.com/aws/aws-cdk/pull/19569 to get merged in.